group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetMetadataManager.java

```diff
@@ -1006,7 +1006,7 @@ public boolean cleanupExpiredOffsets(String groupId, List<CoordinatorRecord> rec
         TimelineHashMap<String, TimelineHashMap<Integer, OffsetAndMetadata>> offsetsByTopic =
             offsets.offsetsByGroup.get(groupId);
         if (offsetsByTopic == null) {
-            return true;
+            return !openTransactions.contains(groupId);
```

Comment:
nit: Should we update the @return in the javadoc to state that we return false if the group has pending transactions?